src/StoreMeta.h

```diff
@@ -80,67 +64,96 @@ enum {
+/// limit protects code that adds individual swap metadata field sizes from
+/// overflowing and also prevents allocation of huge buffers when loading
+/// variable-length fields. Reevaluate this limit when increasing MAX_URL.
+const size_t SwapMetaFieldValueLengthMax = 64*1024;
+
+// TODO: Move to src/store/SwapMetaReading or similar, along with StoreMetaUnpacker.
+/// a swap metadata field inside the buffer given to StoreMetaUnpacker
+class SwapMetaView
 {
-protected:
-    StoreMeta() : length(-1), value(nullptr), next(nullptr) { }
-    StoreMeta(const StoreMeta &);
-    StoreMeta& operator=(const StoreMeta &);
+public:
+    /// ensures that our fixed-size field value has the given expected length
+    void checkExpectedLength(size_t) const;
 
 public:
-    static bool validType(char);
-    static int const MaximumTLVLength;
-    static int const MinimumTLVLength;
-    static StoreMeta *Factory(char type, size_t len, void const *value);
-    static StoreMeta **Add(StoreMeta **tail, StoreMeta *aNode);
-    static void FreeList(StoreMeta **head);
-
-    virtual char getType() const = 0;
-    virtual bool validLength(int) const;
-    virtual bool checkConsistency(StoreEntry *) const;
-    virtual ~StoreMeta() {}
-
-    int length;
-    void *value;
-    tlv *next;
+    /// A serialized rawLength-byte value (i.e. V in swap meta TLV field).
+    /// The value contents may be completely malformed/bogus.
+    /// Not meaningful when type is STORE_META_VOID.
+    const void *rawValue = nullptr;
+
+    /// The number of bytes in rawValue (i.e. L in swap meta TLV field).
+    /// This length may not match the length of valid fields of this type.
+    /// Not meaningful when type is STORE_META_VOID.
+    size_t rawLength = 0;
+
+    /// The serialized type (i.e. T in swap meta TLV field).
+    /// This type value may not match any named by SwapMetaType.
+    Store::RawSwapMetaType rawType = 0;
+
+    /// The sanitized TLV type that always matches one named by SwapMetaType:
+    /// rawType (if matches a value named by SwapMetaType) or STORE_META_VOID.
+    SwapMetaType type = STORE_META_VOID;
+
+private:
+    /*
+     * These construction/copying methods are private so that no outside code
+     * can create/own SwapMetaView objects but SwapMetaIterator. Others loop
+     * StoreMetaUnpacker for read-only access to the current view of metadata.
+     */
+    friend class SwapMetaIterator;
+
+    SwapMetaView() = default;
+    SwapMetaView(const SwapMetaView &) = default;
+    SwapMetaView(SwapMetaView &&) = default;
+    SwapMetaView &operator =(const SwapMetaView &) = default;
+    SwapMetaView &operator =(SwapMetaView &&) = default;
+
+    /// positions the view at the first swap meta field in the given buffer
+    /// \param begin is where the buffer and the field starts
+    /// \param end is where the buffer (but not necessarily the field) finishes
+    explicit SwapMetaView(const void *begin, const void * const end);
```

Comment:
Please bring up the style policy change for discussion in squid-dev.

Pending that discussion I am going to retract this change request. Please d not take it as agreement with the layout style used here. The file is likely to be re-shuffled later by the ongoing class style cleanup project.

Reply:
> Please bring up the style policy change for discussion in squid-dev.

```
Jun 21 20:45:16 UTC: ...
to=<squid-dev@lists.squid-cache.org>,
relay=lists.squid-cache.org[104.130.201.120]:25, ...
status=sent (250 2.0.0 Ok: queued as 81C45E00C5)
```


Done. (I cannot share a link to my post because, AFAICT, the post is currently stuck somewhere on the mail server. I do not have access to check the details, but it was accepted by the SMTP agent and is not in the moderation queue. @kinkie, if you can check the squid-dev mailing list server, please do.)

To avoid misunderstanding, please do not misinterpret that RFC posting as an implicit agreement that this PR has violated some policy (that may now be changed to accommodate this PR and similar cases). We still apparently disagree on what the style status quo is, but that disagreement should not prevent us from making progress with that RFC and this PR.

> Please d not take it as agreement with the layout style used here

Sure, I will not.